src/pykeen/evaluation/evaluator.py

```diff
@@ -416,10 +385,78 @@ def _check_slicing_availability(model: Model, batch_size: int, entities: bool, r
             )
 
 
+def _hasher(kwargs: Mapping[str, Any]) -> int:
+    """Share optimal batch size whenever this hash matches."""
+    return hash((id(kwargs["evaluator"]), kwargs["mapped_triples"].shape[0], kwargs["targets"]))
+
+
+@maximize_memory_utilization(parameter_name=("batch_size", "slice_size"), hasher=_hasher)
+@torch.inference_mode()
+def optimized_evaluate(
```

Comment:
can this take the place of the previous `evaluate` function?

Reply:
Yes; it should maybe get the same name? 🤔

Reply:
done in f476c94